src/SFA.DAS.Tools.Servicebus.Support.Web/Controllers/MessageListController.cs

```diff
@@ -36,58 +33,69 @@ public class MessageListController : Controller
             IQueryHandler<GetMessagesByIdQuery, GetMessagesByIdQueryResponse> getMessagesByIdQuery,
             IQueryHandler<GetQueueDetailsQuery, GetQueueDetailsQueryResponse> getQueueDetailsQuery,
             IMessageService messageService,
-            ICommandHandler<SendMessagesCommand, SendMessagesCommandResponse> sendMessagesCommand,
-            ICommandHandler<DeleteQueueMessagesCommand, DeleteQueueMessagesCommandResponse> deleteQueueMessageCommand,
-            IConfiguration config)
+            IConfiguration config,
+            IUserSessionService userSessionService)
         {
             _userService = userService;
             _getMessagesQuery = getMessagesQuery;
             _getMessagesByIdQuery = getMessagesByIdQuery;
             _getQueueDetailsQuery = getQueueDetailsQuery;
             _messageService = messageService;
-            _sendMessagesCommand = sendMessagesCommand;
-            _deleteQueueMessageCommand = deleteQueueMessageCommand;
+            _userSessionService = userSessionService;
             _errorQueueRegex = config["ErrorQueueRegex"];
+
         }
 
         public async Task<IActionResult> Index()
         {
-            var response = await _getMessagesQuery.Handle(new GetMessagesQuery() 
-            { 
+            var response = await _getMessagesQuery.Handle(new GetMessagesQuery()
+            {
                 UserId = _userService.GetUserId(),
                 SearchProperties = new SearchProperties
                 {
                     Offset = 0,
                     Limit = 1
                 }
             });
-            
+
+            if (response?.Count == 0)
+            {
+                await DeleteUserSession();
+                return RedirectToAction("Index", "Servicebus");
+            }
+
+            var queueName = GetQueueName(response.Messages);
+            await _userSessionService.UpsertUserSession(queueName);
+
             return View(new MessageListViewModel()
             {
                 Count = response.Count,
                 QueueInfo = (await _getQueueDetailsQuery.Handle(new GetQueueDetailsQuery()
                 {
-                    QueueName = GetQueueName(response.Messages)
-                })).QueueInfo
+                    QueueName = queueName
+                })).QueueInfo,
+                UserSession = await _userSessionService.GetUserSession()
+
             });
         }
 
         public async Task<IActionResult> ReceiveMessages(string queue)
         {
+            await _userSessionService.UpsertUserSession(queue);
```

Comment:
Could updating the Session be done as middleware, that way we wouldn't need to explicitly do it on every call?

Reply:
@DannyGB done